### DISCUSSION TITLE
Access and secret key format and generation

### DISCUSSION THREAD

**Discussion by toastie89** [minio/minio#15296] [Q&A] [answered]
Hi! So far I've generated access and secret key as random 30 character string containing letters and numbers using `pwgen --secure --capitalize --numerals --ambiguous 30`.

I couldn't find any definition about the format and length of these keys. Is there some documentation or best practice about the allowed characters and maximum length?

Thanks in advance for your guidance!

**Comment by harshavardhana** [comment, marked as answer]
We only enforce length https://github.com/minio/minio/blob/master/internal/auth/credentials.go#L35

**Reply by toastie89** [reply]
Thanks for your prompt reply! Is there any guidance / best practices about the maximum length or on characters to use or avoid?
